components/script/dom/rtcpeerconnection.rs

```diff
@@ -288,7 +292,7 @@ impl RTCPeerConnection {
         event.upcast::<Event>().fire(self.upcast());
     }
 
-    fn on_data_channel_event(&self, channel_id: DataChannelId, event: DataChannelEvent) {
```

Comment:
can we not just use self.global() here?

Reply:
That triggers the [same panic](https://github.com/servo/servo/issues/27037#issue-643154948). Passing `global` to that function was just an attempt to workaround the panic.